ph5/utilities/set_deploy_pickup_times.py

```diff
@@ -85,91 +83,199 @@ def get_args():
+                if dep_time[inc] is None:
+                    of.write("\t%s\n" % line)
+                else:
+                    of.write("\tdeploy_time/ascii_s=%s\n" %
+                             dep_time[inc].ascii_s)
         elif pickupRE.match(line):
             key, val = line.split('=')
             pre, post = key.split('/')
             post = post.strip()
             if post == 'epoch_l':
-                of.write("\tpickup_time/epoch_l = %d\n" % pu_time.epoch_l)
+                if pu_time[inc] is None:
+                    of.write("\t%s\n" % line)
+                else:
+                    of.write("\tpickup_time/epoch_l=%d\n" %
+                             pu_time[inc].epoch_l)
             elif post == 'micro_seconds_i':
-                of.write("\tpickup_time/micro_seconds_i = %d\n" %
-                         pu_time.micro_seconds_i)
+                if pu_time[inc] is None:
+                    of.write("\t%s\n" % line)
+                else:
+                    of.write("\tpickup_time/micro_seconds_i=%d\n" %
+                             pu_time[inc].micro_seconds_i)
             elif post == 'type_s':
-                of.write("\tpickup_time/type_s = %s\n" % pu_time.type_s)
+                if pu_time[inc] is None:
+                    of.write("\t%s\n" % line)
+                else:
+                    of.write("\tpickup_time/type_s=%s\n" % pu_time[inc].type_s)
+                if auto is True:
+                    inc = inc+1  # Increments the list of input units.
+                else:
+                    inc = 0
             elif post == 'ascii_s':
-                of.write("\tpickup_time/ascii_s = %s\n" % pu_time.ascii_s)
+                if pu_time[inc] is None:
+                    of.write("\t%s\n" % line)
+                else:
+                    of.write("\tpickup_time/ascii_s=%s\n" %
+                             pu_time[inc].ascii_s)
         else:
             of.write("\t%s\n" % line)
 
 
+def kefTables(kef_file, NICKNAME, PATH):
+
+    k = kefx.Kef(ARRAY_FILE)
+    k.open()
+    while True:
+        n = k.read(10000)
+        if n == 0:
+            LOGGER.error("Kef file is empty")
+            break
+        # Get Das_g references
+        ret = k.strip_receiver_g()
+        if ret:
+            add_references(ret)
+
+            # Make sure Array_t_xxx, Event_t_xxx, and Offset_t_aaa_sss exist
+            arrays, events, offsets = k.strip_a_e_o()
+
+
 def main():
-    global ARRAY_FILE, DEPLOY, PICKUP
+    global ARRAY_FILE, DEPLOY, PICKUP, AUTO_CORRECT, NICKNAME, PATH
 
     get_args()
-
+    dep_time = []
+    pu_time = []
     if not os.path.exists(ARRAY_FILE):
         LOGGER.error("Can't open {0}!".format(ARRAY_FILE))
         sys.exit()
     else:
         fh = open(ARRAY_FILE)
         mdir = os.path.dirname(ARRAY_FILE)
         base = os.path.basename(ARRAY_FILE)
-        base = '_' + base
+        base = 'autocortime_{0}'.format(base)
```

Comment:
I wonder if we should have a check to see if a autocortime_*.kef file already exists? This way the program doesn't overwrite anything.

Reply:
Commits 896cee4